src/main/java/com/google/api/codegen/configgen/transformer/RetryTransformer.java

```diff
@@ -27,12 +28,23 @@
   public static final String RETRY_CODES_NON_IDEMPOTENT_NAME = "non_idempotent";
   public static final String RETRY_PARAMS_DEFAULT_NAME = "default";
 
-  public void generateRetryDefinitions(InterfaceView.Builder interfaceView) {
-    interfaceView.retryCodesDef(generateRetryCodes());
+  public void generateRetryDefinitions(
+      InterfaceView.Builder interfaceView, TransportProtocol transportProtocol) {
+    interfaceView.retryCodesDef(generateRetryCodes(transportProtocol));
     interfaceView.retryParamsDef(generateRetryParams());
   }
 
-  private List<RetryCodeView> generateRetryCodes() {
+  private List<RetryCodeView> generateRetryCodes(TransportProtocol transportProtocol) {
+    switch (transportProtocol) {
+      case HTTP:
+        return generateRetryCodesHttp();
+      case GRPC:
+      default:
+        return generateRetryCodesGrpc();
+    }
+  }
+
+  private List<RetryCodeView> generateRetryCodesGrpc() {
```

Comment:
I'd prefer to not have grpc / http specific logic in this general `RetryTransformer` class if it's not too difficult.

Reply:
Done.